llvm/docs/AIToolPolicy.md

```diff
@@ -0,0 +1,154 @@
+  barriers is a welcome and encouraged practice, but keep in mind, we value your
+  unique voice and perspective.
+
+* **Limit AI Tools for Reviewing.** As with creating code, documentation, and
+  other contributions, reviewers may use AI tools to assist in providing
+  feedback, but not to wholly automate the review process. Particularly, AI
+  should not make the final determination on whether a contribution is accepted
+  or not. The same principle of ownership applies to review comment
+  contributions as it does to code contributions.
+
+[ai-slop]: https://simonwillison.net/2024/May/8/slop/
+
+This policy extends beyond code contributions and includes, but is not limited
+to, the following kinds of contributions:
+
+- Code, usually in the form of a pull request
+- RFCs or design proposals
+- Issues or security vulnerabilities
+- Comments and feedback on pull requests
+
+
+## Extractive Changes
+
+Sending patches, PRs, RFCs, comments, etc to LLVM, is not free -- it takes a
+lot of maintainer time and energy to review those contributions! We see the act
+of sending low-quality, un-self-reviewed contributions to the LLVM project as
+"extractive." It is an attempt to extract work from the LLVM project community
+in the form of review comments and mentorship, without the contributor putting
+in comensurate effort to make their contribution worth reviewing.
+
+Our **golden rule** is that a contribution should be worth more to the project
+than the time it takes to review it. These ideas are captured by this quote
+from the book [Working in Public][public] by Nadia Eghbal:
+
+[public]: https://press.stripe.com/working-in-public
+
+> \"When attention is being appropriated, producers need to weigh the costs and
+> benefits of the transaction. To assess whether the appropriation of attention
+> is net-positive, it's useful to distinguish between *extractive* and
+> *non-extractive* contributions. Extractive contributions are those where the
+> marginal cost of reviewing and merging that contribution is greater than the
+> marginal benefit to the project's producers. In the case of a code
+> contribution, it might be a pull request that's too complex or unwieldy to
+> review, given the potential upside.\" \-- Nadia Eghbal
+
+We encourage contributions that help sustain the project. We want the LLVM
+project to be welcoming and open to aspiring compiler engineers who are willing
+to invest time and effort to learn and grow, because growing our contributor
+base and recruiting new maintainers helps sustain the project over the long
+term. We therefore automatically post a greeting comment to pull requests from
+new contributors and encourage maintainers to spend their time to help new
+contributors learn.
+
+## Handling Violations
+
+If a maintainer judges that a contribution is *extractive* (i.e. it is
+generated with tool-assistance or simply requires significant revision), they
```

Comment:
I believe these lines don't capture the intent of the policy. It suggests any contribution generated with tool assistance is extractive. Perhaps "it is generated with tool-assistance in a way inconsistent with our AI tool usage policy"?